src/test/java/uk/gov/hmcts/reform/civil/service/robotics/mapper/AddressLinesMapperTest.java

```diff
@@ -110,20 +110,64 @@ void shouldReturnOriginalAddress_whenSplittingSpreadsOverThreeLines() {
 
             Address result = mapper.splitLongerLines(address);
 
-            assertThat(result).isEqualTo(address);
+            assertThat(result.getAddressLine1()).isEqualTo(address.getAddressLine1());
+            assertThat(result.getAddressLine2()).isEqualTo(address.getAddressLine2());
+            assertThat(result.getAddressLine3()).isEqualTo("abcdefghijk12345678901234567890");
+            assertThat(result.getPostTown()).isEqualTo("zxcvbnmzxcvbnm");
+        }
+
+        @Test
+        void shouldSplitAddressBySpace_whenSplittingSpreadsOverThreeLinesWithSpaces() {
+            Address address = Address.builder()
+                .addressLine1("1234567890 1234567890")
+                .addressLine2("I am the    second   line")
```

Comment:
Question, there are more than three lines with spaces here, why isn't it split into addressline3?

Reply:
the split only happens if the length of line plus any carryover from previous lines' split is greater than limit, i.e. 35 chars. that's why addressline1 and addressLine2 remain as is. addresslines[2] during the logic contains "abcdefghijk 1234567890 1234567890 1234567899" (the post comma stuff 'zxcvbnmzxcvbnm' goes to addresslines[3] is more than 35 chars). So we accommodate as much as 35 chars and take the spillover to posttwon. notably, RPA fields only have addresslines 1-5 and postcode, i.e no distinction between county, town, country etc.